fix categories crud

gilly7 opened the pull request:
Fixed and enhanced Categories CRUD operations #12 

- Implemented proper error handling for all CRUD operations
- Added confirmation dialog for delete action
- Enhanced create and update operations with input validation
- Improved error messages and success feedback
- Refactored state management for better performance

![Screenshot 2025-01-19 003129](https://github.com/user-attachments/assets/e23e8b53-0228-4fde-957e-c839ca7dae28)
![Screenshot 2025-01-19 003228](https://github.com/user-attachments/assets/82e2a50e-de59-44ac-998c-58e23256e301)

gilly7 committed:
fix categories crud

Korirhillary reviewed (approved):
Great improvements .

etowett commented on backend/category_routes.py:
```diff
@@ -76,18 +76,48 @@ def read_category(category_id: int, db: Session = Depends(get_db)):
     
-    return category
+    # Calculate post_count
+    post_count = db.query(Post).filter(Post.category_id == category.id).count()
```
do we really need to do this on category update? maybe not

etowett commented on frontend/app/lib/api.ts:
```diff
@@ -34,16 +34,21 @@ export const getCategories = async (page: number = 1, limit: number = 10) => {
     body: JSON.stringify(data),
   });
+
+  if (!response.ok) {
+    const errorData = await response.json();
+    throw new Error(errorData.detail || 'Failed to update category');
+  }
```
We already handle this section in the `fetchWithAuth` function, please remove it

gilly7 replied:
These have been handle on this commit 2976fe9e7be8099ee7942ebed26e8292bc3ac15d

etowett reviewed (approved):
All good but check the comments

gilly7 committed:
Refactor update_category endpoint to remove redundant post_count calculation

- Removed the calculation of post_count during category updates.
- Simplified the response by returning the updated category directly.
- Improved performance by avoiding unnecessary database queries.
